### ISSUE TITLE
invalidationBehavior: 'delayed' behaviour for sequential mutations

### ISSUE THREAD

**Issue by HarlesPilter** [reduxjs/redux-toolkit#3979]
I have created a [codesandbox ](https://codesandbox.io/p/devbox/m7jsmv?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clq6dvvup0006356riqacodnu%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clq6dvvup0002356ri8wkteim%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clq6dvvup0003356rm4jh0v2p%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clq6dvvup0005356r4qinyudg%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clq6dvvup0002356ri8wkteim%2522%253A%257B%2522id%2522%253A%2522clq6dvvup0002356ri8wkteim%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clq6dvvup0005356r4qinyudg%2522%253A%257B%2522id%2522%253A%2522clq6dvvup0005356r4qinyudg%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522port%2522%253A3000%252C%2522id%2522%253A%2522clq6eyyws008f356rq4loafqv%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clq6eyyws008f356rq4loafqv%2522%257D%252C%2522clq6dvvup0003356rm4jh0v2p%2522%253A%257B%2522id%2522%253A%2522clq6dvvup0003356rm4jh0v2p%2522%252C%2522activeTabId%2522%253A%2522clq6eyyta006t356rtt5vtoaf%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%252C%2522id%2522%253A%2522clq6eyyta006t356rtt5vtoaf%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522id%2522%253A%2522clq6f39se00m6356rlqjc4p4q%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clq6gjdfi0009egi6184h9glh%2522%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522CSB_RUN_OUTSIDE_CONTAINER%253D1%2520devcontainer%2520templates%2520apply%2520--template-id%2520%255C%2522ghcr.io%252Fdevcontainers%252Ftemplates%252Ftypescript-node%255C%2522%2520--template-args%2520%27%257B%257D%27%2520--features%2520%27%255B%255D%27%2522%252C%2522id%2522%253A%2522clq6gtvxa030c356r15vsyd3k%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A16.35416666666667%257D) to display the problem. 

TLDR: Two sequentially fired mutations that invalidate the same two tags, makes second invalidation wait for first queries to finish.

I have 3 cases displayed there (last one behaves weirdly):
1. Press the button "Run 1 mutation". Mutation is fired and when it succeeds, it invalidates 2 tags, that in-turn fire short and long queries. Thats all good 👍
![image](https://github.com/reduxjs/redux-toolkit/assets/16189897/33043ca6-ca73-4b4d-812f-95f09b70add7)
2. Press the button "Run 2 mutations in parallel". Mutation 1 (shorter) and Mutation 2(longer) are fired at the same time.
![image](https://github.com/reduxjs/redux-toolkit/assets/16189897/9ca6cc4f-b5b6-4d43-8110-321de849da54)
They both invalidate the same 2 tags. You can see in the network tab, that the "short" and "long" query fire only when the longer mutation has finished. Thats all good 👍
![image](https://github.com/reduxjs/redux-toolkit/assets/16189897/68ba90a0-bc4b-4f3a-9d56-f778a378d245)
4. Press the button "Run 2 mutations sequentially". 
![image](https://github.com/reduxjs/redux-toolkit/assets/16189897/8878f409-d3ab-4b9d-8ca0-90d41d6da995)
The second mutation is called when the first one is completed. But because the first mutation is completed the two tags are invalidated and "short" and "long" queries are also fired. Even when the second mutation completes, it waits for both "short" and "long" query to finish, to invalidate its tags and re-fire "short" and "long" queries to get the latest state from BE. Thats bad 👎
![image](https://github.com/reduxjs/redux-toolkit/assets/16189897/2edc9b26-b181-4544-8bc6-6ce7b81f0941)

I think the current behaviour for sequential mutation is not correct.
Options how it could be better:

- ![Option1](https://github.com/reduxjs/redux-toolkit/assets/16189897/43c28fa2-a617-46bf-a666-3214a20f4087)
- ![Option2](https://github.com/reduxjs/redux-toolkit/assets/16189897/fa3ad3de-3c03-42bf-99f4-f0683604c1a7)
- ![Option3](https://github.com/reduxjs/redux-toolkit/assets/16189897/88b2d78a-752b-4571-ae5b-6ea00ef85a0a)
